modules/end-user-portal/src/Pages/Apply/Index.cshtml.cs

```diff
@@ -0,0 +1,23 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.Config;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.Data.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Pages.Apply;
+
+public class IndexModel : BaseModel
+{
+    private readonly ITempDataProviderService _tempDataProviderService;
+
+    public IndexModel(IOptions<ApplicationParameters> applicationParameters, ITempDataProviderService tempDataProviderService) : base(applicationParameters)
+    {
+        _tempDataProviderService = tempDataProviderService;
+    }
+
+    public IActionResult OnGetAsync()
+    {
+        _tempDataProviderService.RemoveAll();
```

Comment:
Do we want to do this OnGet of the Page, maybe this should be done on the clicking of the Apply Button to start a new application?

Reply:
Makes sense - I've moved the logic.